### ISSUE TITLE
`next/image` external domains no longer working

### ISSUE THREAD

**Issue by segersniels** [vercel/next.js#23590]
### What version of Next.js are you using?

10.1.2

### What version of Node.js are you using?

15.8.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next dev

### Describe the Bug

Since version `10.1.x` the external image domains are no longer working under `next.config.js`. Take the following basic example.

```js
module.exports = {
  images: {
    domains: ['cdn.sanity.io', 'pbs.twimg.com'],
  },
};
```

Error:
```
Error: Invalid src prop (https://cdn.sanity.io/images/foo/production/53ce1c173eb7cc38b81104e85336d5bf3bc3c864-1004x733.png?w=2000&h=2000&fit=max) on `next/image`, hostname "cdn.sanity.io" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host
```

### Expected Behavior

Next.js `next/image` is able to render and optimise external images defined under the `images` section in `next.config.js`

### To Reproduce

Run the most basic example that you can and try to render an image from an external domain like twitter using `next/image`.

**Comment by felipefcm** [comment]
Reproducible in Linux.

**Comment by gdowmont** [comment]
Reproducible on MacOS and NodeJS 14.x

**Comment by gdowmont** [comment]
Looks like similar issue has been linked in https://github.com/vercel/next.js/issues/23523#issuecomment-813393374

Passing `unoptimized={true}` allows images to render

**Comment by Tobi-mmt** [comment]
I have the same issue with next 10.1.3, node 14.15.4

**Comment by chenrui333** [comment]
I have the same issue on both 10.1.3 and 10.2.0

**Comment by rheng001** [comment]
Facing same issue

**Comment by segersniels** [comment]
Update: it does seem to work in a production environment (eg. deployed to Vercel) but not in a local environment.

**Comment by chenrui333** [comment]
This issue does prevent us from upgrading to next 10.2.0 (or 10.1.3), can we bump the priority for this bug? Thanks!

**Comment by intensr** [comment]
any update on this?

**Comment by rheng001** [comment]
Still having to pass unoptimized={true} for local environment :(

**Comment by jflayhart** [comment]
@dstapleton92 @timneutkens what's strange is if I do `NODE_ENV=test yarn start` it's fine... so is Jest not configured properly to pull in the `next.config.js` file? Let me try something...

**Comment by jflayhart** [comment]
found a workaround in Jest specs 😂 

```
beforeEach(() => {
  process.env.NODE_ENV = 'production'
})

afterEach(() => {
  process.env.NODE_ENV = 'test'
})
```

**Comment by jflayhart** [comment]
@timneutkens this also works:

```
<Image
        src={marketing_hero}
        width={1080}
        height={550}
        unoptimized={process.env.NODE_ENV === 'test'}.  // <------ THIS 
      />
```

**Comment by intensr** [comment]
noob idea: could it be something sanity / cdn specific, since an external source of error could've changed over night. and tim tested with pbs.twimg.com rather than sanity.

**Comment by timneutkens** [comment]
@felipefcm `The requested resource isn't a valid image.`, this is related to the url you're linking to not returning the correct MIME type, so your issue is related to #23523.

@jflayhart the initial issue does not mention Jest anywhere, your issue sounds like #21549.

**Comment by votemike** [comment]
I am also having this problem. Again, only when running tests (I'm using Jest and RTL). It works fine locally and when deployed to Vercel.

**Comment by segersniels** [comment]
Sorry for not replying anytime sooner but my original issue has been resolved. 

I was using a wrapper package that I wrote that was using an older version of `next/image` which seemingly stopped working since the latest update because of a mismatch between next versions. Updating the dependency's next version resolved the issue. Unsure what originally caused the issue but so far no longer experiencing any issues.

**Comment by imBAOSS** [comment]
I just had this problem and managed to solve it. What happened for me was the actual import used. I believe the right import you want to use is with the lowercased version of the "i" in image as opposed to uppercase.

`import Image from 'next/Image` gave me the invalid src issues not being configured in `next.config.js` even though it was. The `unoptimized={true}` param workaround did work for me, but the solution I was looking for was importing with a lowercase i in Image like below:

`import Image from 'next/image` fixed the issue for me. Importing like this makes the `<Image />` component work as expected.

**Comment by timneutkens** [comment]
Fixed in #26502

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
